core/services/ocr2/plugins/threshold/plugin.go

```diff
@@ -0,0 +1,99 @@
+package threshold
+
+import (
+	"encoding/json"
+	"fmt"
+
+	"github.com/pkg/errors"
+
+	"github.com/smartcontractkit/libocr/commontypes"
+	libocr2 "github.com/smartcontractkit/libocr/offchainreporting2plus"
+
+	decryptionPlugin "github.com/smartcontractkit/tdh2/go/ocr2/decryptionplugin"
+	"github.com/smartcontractkit/tdh2/go/tdh2/tdh2easy"
+
+	"github.com/smartcontractkit/chainlink/v2/core/services/job"
+)
+
+type ThresholdServicesConfig struct {
+	DecryptionQueue    decryptionPlugin.DecryptionQueuingService
+	KeyshareWithPubKey []byte
+	ConfigParser       decryptionPlugin.ConfigParser
+}
+
+func NewThresholdService(sharedOracleArgs *libocr2.OCR2OracleArgs, conf *ThresholdServicesConfig) (job.ServiceCtx, error) {
+	fmt.Println("Private Key Share With Public Key: ", string(conf.KeyshareWithPubKey))
+
+	publicKey, privKeyShare, err := UnmarshalKeys(conf.KeyshareWithPubKey)
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to unmarshal threshold key share with public key")
+	}
+
+	// It can be safely assumed all oracle nodes have been given the correct key share,
+	// so we use the key share index as the oracle's index.
+	// If an oracle was given the wrong key share, key share decryption would have failed.
+	oracleToKeyShare := make(map[commontypes.OracleID]int)
+	oracleToKeyShare[commontypes.OracleID(privKeyShare.Index())] = privKeyShare.Index()
```

Comment:
Looks like the reporting plugin expects a full map and you're  only setting one key.
What are those indices for, anyway? Isn't this an identity mapping  anyway (map[x] = x)?
@stchrysa what's the purpose of oracleToKeyShare?